src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/service/GenerateCoverSheetService.java

```diff
@@ -99,11 +105,39 @@ private void prepareCoverSheet(CaseDetails caseDetails, String partyAddressCcdFi
                     : (Map) caseData.get(partyAddressCcdFieldName)))
                 .build();
             caseData.put(ADDRESSEE, addressee);
-            caseData.put(CTSC_CONTACT_DETAILS, buildCtscContactDetails());
+            caseData.put(COURT_CONTACT_DETAILS, getCourtContactDetails(caseDetails));
             caseData.put(CASE_NUMBER, nullToEmpty(caseDetails.getId()));
         }
     }
 
+    private String getCourtContactDetails(CaseDetails caseDetails) {
+        /*
+        if (isContestedApplication(caseDetails)) {
+            // will implement for FRC Contact Details
+        }
+        */
+        return formatCtscContactDetailsForCoversheet();
+    }
+
+    private String formatCtscContactDetailsForCoversheet() {
+        CtscContactDetails coversheetCtscContactDetails = CtscContactDetails.builder()
+            .serviceCentre("HMCTS Financial Remedy")
+            .poBox("PO BOX " + CTSC_PO_BOX)
+            .town(CTSC_TOWN)
+            .postcode(CTSC_POSTCODE)
+            .build();
+
+        return Stream.of(coversheetCtscContactDetails.getServiceCentre(),
+                coversheetCtscContactDetails.getPoBox(),
+                coversheetCtscContactDetails.getTown(),
+                coversheetCtscContactDetails.getPostcode())
+            .filter(Objects::nonNull)
+            .map(Object::toString)
+            .filter(StringUtils::isNotEmpty)
+            .filter(s -> !s.equals("null"))
```

Comment:
I would probably avoid doing this check, we already have the null check above, if they type null into a text box then fair play to them